### PR TITLE
feat: Story 48.2 - Calculate split ratio from open position and CSV data

### LINKED ISSUE #929: Story 48.2: Calculate Split Ratio from Open Position and CSV Data
## Story

As a trader,
I want the split ratio to be automatically derived from existing position data and the CSV post-split quantity,
so that the correct adjustment factor is applied without manual input.

## Acceptance Criteria

1. **Given** a reverse split for OXLC where the CSV post-split quantity is 306 and the total current open position quantity across all lots is 1530, **When** the ratio is calculated as `1530 / 306`, **Then** the system determines a ratio of 5 (i.e., a 1-for-5 reverse split).
2. **Given** a forward split where the current open quantity is 100 and the CSV post-split quantity is 200, **When** the ratio is calculated as `100 / 200 = 0.5`, **Then** the system correctly identifies it as a 2-for-1 forward split.
3. **Given** no open lots exist for the split symbol at the time of processing, **When** the ratio calculation is attempted, **Then** the split row is skipped with a warning logged and no exception is thrown.
4. **Given** all changes, **When** `pnpm all` runs, **Then** all unit tests (including new unit tests for ratio calculation) pass.

## Story File
`_bmad-output/implementation-artifacts/48-2-calculate-split-ratio.md`

### PR DESCRIPTION
**Pull request by DaveMBush**
## Summary

Implements `calculateSplitRatio()` service function that automatically derives the stock split ratio from the trader's existing open position and the post-split quantity in the CSV import row.

**Formula**: `ratio = totalCurrentOpenQuantity / csvPostSplitQuantity`
- ratio > 1 → reverse split (e.g. 5 means 1-for-5 reverse split)
- ratio < 1 → forward split (e.g. 0.5 means 2-for-1 forward split)

## Changes

- **New**: `calculate-split-ratio.function.ts` — queries open trades by symbol, sums quantities, returns ratio or `null`
- **New**: `calculate-split-ratio.function.spec.ts` — 7 unit tests covering all acceptance criteria
- **Fix**: `fidelity-data-mapper.function.spec.ts` — adds missing in-lieu row skip test (pre-existing coverage gap from story 48.1)

## Testing

All tests pass:
- `pnpm all` (lint + build + unit tests with 100% coverage)
- `pnpm e2e:dms-material:chromium` — 620 passed
- `pnpm e2e:dms-material:firefox` — 623 passed
- `pnpm dupcheck` — 0 duplicates
- `pnpm format` — no changes needed

Closes #929

## Summary by CodeRabbit

* **New Features**
  * Added automatic stock split ratio calculation for imported transactions.

* **Bug Fixes**
  * Improved handling of fractional-share Fidelity imports to skip zero-quantity records.

* **Tests**
  * Added comprehensive tests for split ratio calculations across edge cases.
  * Added test validating fractional-share import filtering.

### CODE REVIEW

File: apps/server/src/app/routes/import/calculate-split-ratio.function.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+ * Formula: ratio = totalCurrentOpenQuantity / csvPostSplitQuantity
+ * - ratio > 1 → reverse split (e.g., 5 means 1-for-5 reverse split)
+ * - ratio < 1 → forward split (e.g., 0.5 means 2-for-1 forward split)
+ *
+ * Returns null if no open lots exist for the symbol (caller must skip the split row).
+ */
+export async function calculateSplitRatio(
+  symbol: string,
+  csvPostSplitQuantity: number
+): Promise<number | null> {
```
[the remark this answers is not in the document]
Comment (DaveMBush) [reply apps/server/src/app/routes/import/calculate-split-ratio.function.ts]: Addressed in commit a2d6f85. The `!Number.isFinite(csvPostSplitQuantity) || csvPostSplitQuantity <= 0` guard was added at the top of the function (lines 22-27) before any Prisma calls or division, making the divisor-safety check unconditional. The 4 new unit tests for zero, negative, NaN, and Infinity inputs confirm the guard works correctly.
